### PR TITLE
Code style improvements.

### PR DESCRIPTION
**Pull request by cpojer**
Planning to use this at FB. Fixing up some of the code style: 80 char limit, semicolons.

I noticed that this git repo is a little inconsistent with semicolons, so I hope it is not controversial to add them everywhere.

### CODE REVIEW

**Comment by cpojer** [comment]
there is a few more missing ones, fixing now.

File: src/core.js [hunk 1]
```diff
@@ -85,25 +87,30 @@ Promise.prototype._handle = function(deferred) {
...
+      deferred.promise._reject(LAST_ERROR);
     } else {
-      deferred.promise._resolve(ret)
+      deferred.promise._resolve(ret);
     }
   });
 };
 Promise.prototype._resolve = function(newValue) {
-  //Promise Resolution Procedure: https://github.com/promises-aplus/promises-spec#the-promise-resolution-procedure
+  // Promise Resolution Procedure: http://bit.ly/1IJmVHN
```
Comment (ForbesLindesay) [inline src/core.js]: I'd prefer to keep the full original url, even if it results in a longer line.

**Comment by ForbesLindesay** [comment]
Whenever the `if` breaks onto a new line I want to add `{}`.  i.e.

```js
// good
if (foo) bar();

// good
if (foo) {
  bar();
}

// bad
if (foo)
  bar();
```

**Comment by cpojer** [comment]
done!

File: src/core.js [hunk 2]
```diff
@@ -68,7 +72,8 @@ Promise.prototype._safeThen = function (onFulfilled, onRejected) {
   });
 };
 Promise.prototype.then = function(onFulfilled, onRejected) {
-  if (this.constructor !== Promise) return this._safeThen(onFulfilled, onRejected);
+  if (this.constructor !== Promise)
+    return this._safeThen(onFulfilled, onRejected);
```
Comment (ForbesLindesay) [inline src/core.js]: Needs `{}`

File: src/core.js [hunk 3]
```diff
@@ -85,25 +90,32 @@ Promise.prototype._handle = function(deferred) {
...
+  // Promise Resolution Procedure: https://github.com/promises-aplus/promises-spec#the-promise-resolution-procedure
   if (newValue === this) {
-    return this._reject(new TypeError('A promise cannot be resolved with itself.'))
+    return this._reject(
+      new TypeError('A promise cannot be resolved with itself.')
+    );
   }
-  if (newValue && (typeof newValue === 'object' || typeof newValue === 'function')) {
+  var type = (newValue && typeof newValue);
+  if (type === 'object' || type === 'function') {
```
Comment (ForbesLindesay) [inline src/core.js]: This changes the actual code and is likely to have a performance impact.  Many JS implementations will treat `typeof foo === 'function'` as "Check `foo` is a function" but treat `var t = typeof foo; t === 'function'` as "Get the type of `'foo'` as a string, compare the two strings `t` and `'function'`".  I haven't actually benchmarked this in this particular instance though.

I think it is also clearer to see this as "`newValue` is truthy and of type object or function", I can't come up with an obvious english language description of your code.

**Comment by cpojer** [comment]
done, again! :)

**Pull request merged**